Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -37,29 +38,79 @@ public class RazorTargetTests
+            var root = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var projectSpecificOutDir = Path.Combine(root, "0");
+            var temporaryProjectSpecificOutDir = Path.Combine(root, "0.tmp");
+            var razorSpecificOutDir = Path.Combine(root, "0.Razor");
+            Directory.CreateDirectory(projectSpecificOutDir);
+            Directory.CreateDirectory(temporaryProjectSpecificOutDir);
+            File.WriteAllText(Path.Combine(projectSpecificOutDir, "ProjectInfo.xml"),
+$@"
+<ProjectInfo>
+  <AnalysisSettings>
+    <Property Name = ""sonar.cs.roslyn.reportFilePaths"" > C:\SonarSource\Sandbox\Mixed\Mixed\Main2\bin\Debug\netcoreapp3.1\Main2.dll.RoslynCA.json | C:\SonarSource\Sandbox\Mixed\Mixed\Main2\bin\Debug\netcoreapp3.1\Main2.dll.RoslynCA.json </ Property >
+    <Property Name = ""sonar.cs.analyzer.projectOutPaths"" >{projectSpecificOutDir}</Property>
+  </ AnalysisSettings >
+  <Configuration > Debug </ Configuration >
+  <Platform > AnyCPU </ Platform >
+  <TargetFramework > netcoreapp3.1 </TargetFramework>
+</ProjectInfo> ");
+            File.WriteAllText(Path.Combine(temporaryProjectSpecificOutDir, "ProtoBuf.txt"), string.Empty);
+            var razorSpecificProjectInfo = Path.Combine(razorSpecificOutDir, "ProjectInfo.xml");
+
+            var projectSnippet = $@"
+<PropertyGroup>
+  <SonarTemporaryProjectSpecificOutDir>{temporaryProjectSpecificOutDir}</SonarTemporaryProjectSpecificOutDir>
+  <ProjectSpecificOutDir>{projectSpecificOutDir}</ProjectSpecificOutDir>
+</PropertyGroup>
+";
+
+            var filePath = CreateProjectFile(null, projectSnippet, TargetConstants.RazorSonarCopyProjectInfoFile);
+
+            // Act
+            var result = BuildRunner.BuildTargets(TestContext, filePath, TargetConstants.RazorSonarCopyProjectInfoFile);
+
+            // Assert
+            result.AssertTargetExecuted(TargetConstants.RazorSonarCopyProjectInfoFile);
+            File.Exists(Path.Combine(projectSpecificOutDir, "ProtoBuf.txt")).Should().BeTrue();
+            File.Exists(razorSpecificProjectInfo).Should().BeTrue();
+            File.ReadAllText(razorSpecificProjectInfo).Should().Contain($@"<Property Name = ""sonar.cs.analyzer.projectOutPaths"" >{razorSpecificOutDir}</Property>");
+            Directory.Exists(temporaryProjectSpecificOutDir).Should().BeFalse();
+        }
+
         [TestMethod]
         public void Razor_ForProductProject_CheckErrorLogProperties()
         {
             // Arrange
             var rootInputFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "Inputs");
 
+            var root = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext);
+            var sourceDirectory = Path.Combine(root, "Source");
+            var destinationDirectory = Path.Combine(root, "Source.tmp");
+            Directory.CreateDirectory(sourceDirectory);
+            File.WriteAllText(Path.Combine(sourceDirectory, "ProtoBuf.txt"), string.Empty);
```

Comment:
This whole section can be rewritten with better use of `TestUtils` like `CreateEmptyFile` and additional arguments on `CreatetestSpecificFolderWithsubPaths(TestContet, "Source")`